### PR TITLE
Ch4 misc prefixs for xpmem

### PR DESCRIPTION
**Pull request by minsii**
These small commits are prefixes for the XPMEM PR #3452

### CODE REVIEW

**Comment by minsii** [comment]
test:jenkins/ch4/most

File: src/mpid/ch4/generic/mpidig_recv.h [hunk 1]
```diff
@@ -20,13 +20,17 @@
 #define FUNCNAME MPIDIG_prepare_recv_req
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-static inline int MPIDIG_prepare_recv_req(void *buf, MPI_Aint count, MPI_Datatype datatype,
+static inline int MPIDIG_prepare_recv_req(int rank, int tag, MPIR_Context_id_t context_id,
```
Comment (hajimefu) [inline src/mpid/ch4/generic/mpidig_recv.h]: Actually I'm trying to remove `MPIDIG_prepare_recv_req` in #3590. Do you think this is going to be useful in other places?

Comment (minsii) [reply src/mpid/ch4/generic/mpidig_recv.h]: Yes, I am using it in the XPMEM PR #3452 . I need to create AM receive request in SHM:
https://github.com/pmodels/mpich/blob/8e4e8408244b0e407abd1de38c1ebccb04ca0441/src/mpid/ch4/shm/src/shm_p2p.h#L343-L355

Comment (minsii) [reply src/mpid/ch4/generic/mpidig_recv.h]: @hajimefu Do you think we can go ahead and merge this PR ?

Comment (hajimefu) [reply src/mpid/ch4/generic/mpidig_recv.h]: Yes. I'll drop my patch to remove `MPIDIG_prepare_recv_req`.

**Pull request merged**